pkg/iris/scan.go

```diff
@@ -14,17 +15,16 @@ type Scanner interface {
 }
 
 type scanner struct {
-	api API
+	api   API
+	RunId string
 }
 
-func NewScanner(kind string, uri string) (Scanner, error) {
-	api, err := NewApi(kind, uri)
-	if err != nil {
-		return nil, err
-	}
+func NewScanner(kind string, uri string) Scanner {
+	api := NewApi(kind, uri)
 	return &scanner{
-		api: api,
-	}, nil
+		api:   api,
+		RunId: uuid.New().String(),
```

Comment:
Not sure if the RunId should be an attribute of the scanner. It's ok to leave it like this for now.
A better approach would be to pass it as a parameter when calling the scan() function

Reply:
so you mean we can generate it in `main.go` and then pass it over?

Reply:
correct